quantum/debounce/readme.md

```diff
@@ -1,28 +1,77 @@
+# Debounce algorithms
+
```

Comment:
I don't really like this file. I know it already exists, but I think this would be a great opportunity to let it go.  
I think this content would be better merged with the content of `docs/feature_debounce_type.md` making the content of that file more complete and making the content of this one more accessible.

I'd like to get feedback from the @qmk/collaborators on this topic.

Reply:
Definitely agree with this.  

This info should be moved into the docs, not hidden here.

Reply:
Okay, looks like there's agreement.
I refactored the documentation, and also took the opportunity to add some new information, please see the latest commit I pushed.